Source/Core/Common/CompatPatches.cpp

```diff
@@ -0,0 +1,275 @@
+{
+  watcher->Install({{L"EZFRD64.dll", L"811EZFRD64.DLL"},
+                    [](const LdrDllLoadEvent& event) {
+                      // *EZFRD64 is incldued in software packages for cheapo third-party gamepads
+                      // (and gamepad adapters).
+                      // The module cannot handle its heap being above 4GB, which tends to happen
+                      // very often on modern Windows.
+                      // NOTE: The patch will always be applied, but it will only actually avoid the
+                      // crash if applied
+                      // before module initialization (i.e. called on the Ldr callout path).
+                      auto patcher = ImportPatcher(event.base_address);
+                      patcher.PatchIAT("kernel32.dll", "HeapCreate", HeapCreateLow4GB);
+                    }});
+  watcher->Install({{L"ucrtbase.dll"},
+                    [](const LdrDllLoadEvent& event) {
+                      // ucrtbase implements caching between fseek/fread, old versions have a bug
+                      // such that some reads return incorrect data. This causes noticable bugs
+                      // in dolphin since we use these APIs for reading game images.
+                      Version version;
+                      if (!GetModuleVersion(event.name.c_str(), &version))
+                        return;
+                      const u16 fixed_build = 10548;
+                      if (version.build >= fixed_build)
+                        return;
+                      const UcrtPatchInfo patches[] = {
+                          // 10.0.10240.16384 (th1.150709-1700)
+                          {0xF61ED, 0x6AE7B, 5},
+                          // 10.0.10240.16390 (th1_st1.150714-1601)
+                          {0xF5ED9, 0x6AE7B, 5},
+                          // 10.0.10137.0 (th1.150602-2238)
+                          {0xF8B5E, 0x63ED6, 2},
+                      };
+                      for (const auto& patch : patches)
+                      {
+                        if (ApplyUcrtPatch(event.name.c_str(), patch))
+                          return;
+                      }
+                      // If we reach here, the version is buggy (afaik) and patching failed
+                      auto msg = StringFromFormat(
+                          "You are running %S version %d.%d.%d.%d.\n"
+                          "An important fix affecting dolphin was introduced in build %d.\n"
+                          "You can use dolphin, but there will be known bugs.\n"
+                          "Please update this file by installing the latest Universal C Runtime.\n",
+                          event.name.c_str(), version.major, version.minor, version.build,
+                          version.qfe, fixed_build);
+                      // Use MessageBox for maximal user annoyance
+                      MessageBoxA(nullptr, msg.c_str(), "WARNING: BUGGY UCRT VERSION",
```

Comment:
Currently this code runs during CRT unit. I'm doubtful that translation will work. It's generally best to get the patches in asap, but since for the ucrt one we only really care about getting it done before emulation starts, it could be moved someplace else. Its just not really optimal.